src/main/java/org/broadinstitute/hellbender/utils/FisherExactTest.java

```diff
@@ -21,59 +13,35 @@
 public final class FisherExactTest {
 
     /**
-     * Computes the 2-sided pvalue of the Fisher's exact test.
+     * Computes the 2-sided pvalue of the Fisher's exact test on a normalized table that ensures that the sum of
+     * all four entries is less than 2 * 200.
      */
     public static double twoSidedPValue(final int[][] normalizedTable) {
         Utils.nonNull(normalizedTable);
-        Utils.validateArg(normalizedTable.length == 2, "input must be 2x2 " + Arrays.deepToString(normalizedTable));
-        Utils.validateArg(normalizedTable[0] != null && normalizedTable[0].length == 2, "input must be 2x2 " + Arrays.deepToString(normalizedTable));
-        Utils.validateArg(normalizedTable[1] != null && normalizedTable[1].length == 2, "input must be 2x2 " + Arrays.deepToString(normalizedTable));
+        Utils.validateArg(normalizedTable.length == 2, () -> "input must be 2x2 " + Arrays.deepToString(normalizedTable));
+        Utils.validateArg(normalizedTable[0] != null && normalizedTable[0].length == 2, () -> "input must be 2x2 " + Arrays.deepToString(normalizedTable));
+        Utils.validateArg(normalizedTable[1] != null && normalizedTable[1].length == 2, () -> "input must be 2x2 " + Arrays.deepToString(normalizedTable));
 
         //Note: this implementation follows the one in R base package
         final int[][] x= normalizedTable;
-        final int m = addExact(x[0][0], x[0][1]);
-        final int n = addExact(x[1][0], x[1][1]);
-
-        if (m+n == 0){     //special case, all entries zero
+        final int m = x[0][0] + x[0][1];
+        final int n = x[1][0] + x[1][1];
+        final int k = x[0][0] + x[1][0];
+        final int lo = Math.max(0, k - n);
+        final int hi = Math.min(k, m);
+        final IndexRange support = new IndexRange(lo, hi + 1);
+
+        if (support.size() <= 1){     //special case, support has only one value
             return 1.0;
         }
 
-        final int k = addExact(x[0][0], x[1][0]);
-        final int lo = max(0, k - n);
-        final int hi = min(k, m);
-        final int[] support = range(lo, hi);
         final AbstractIntegerDistribution dist = new HypergeometricDistribution(null, m+n, m, k);
-        final double[] logdc = IntStream.of(support).mapToDouble(i -> dist.logProbability(i)).toArray();
-        final double oddsRatioAtNull = 1.0;
-        final double[] ds = dnHyper(oddsRatioAtNull, logdc, support);
-        final double relErr = 1 + pow(10, -7);
+        final double[] ds = support.mapToDouble(i -> dist.probability(i));
 
-        final double threshold= ds[x[0][0] - lo] * relErr;
+        final double threshold = ds[x[0][0] - lo];
```

Comment:
@meganshand I read your comment about using normalized tables and while I agree with you I don't want this code to rely on that fact.  Fortunately, I think you can get numerical stability with negligible effort:

```java
final double[] logds = support.mapToDouble(i -> dist.logProbability(i));  // probability -> logProbability
final double threshold = ds[x[0][0] - lo];   // no change needed since log is monotonic
final double[] log10ds = DoubleStream.of(logds).filter(d -> d <= threshold).map(MathUtils::logToLog10).toArray();
final double pValue = MathUtils.sumLog10(log10ds);
```